android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncLocalStorageUtil.java

```diff
@@ -142,4 +146,36 @@ private static void deepMergeInto(JSONObject oldJSON, JSONObject newJSON)
       }
     }
   }
+  /**
+   * From Pie and up, Android started to use Write-ahead logging (WAL), instead of journal rollback
+   * for atomic commits and rollbacks.
+   * Basically, WAL does not write directly to the database file, rather to the supporting WAL file.
+   * Because of that, migration to the next storage might not be successful, because the content of
+   * RKStorage might be still in WAL file instead. Committing all data from WAL to db file is called
+   * a "checkpoint" and is done automatically (by default) when the WAL file reaches a threshold
+   * size of 1000 pages.
+   * More here: https://sqlite.org/wal.html
+   * <p>
+   * This helper will force checkpoint on RKStorage, if Next storage file does not exists yet.
+   */
+  public static void verifyAndForceSqliteCheckpoint(Context ctx) {
+    File nextStorageFile = ctx.getDatabasePath("AsyncStorage");
```

Comment:
I'm aware that this leaks next storage DB name, but I think it's harmless here (and db file name does not change much 😄 )

Reply:
Maybe we can make this configurable some time in the future?

Reply:
Yeah, probably something to think about